header-only/layout-builder/awkward/LayoutBuilder.h

```diff
@@ -1151,6 +1153,17 @@ namespace awkward {
         return content_;
       }
 
+      /// @brief Inserts an explicit value in the `index` buffer and
+      /// returns the reference to the builder content.
+      BUILDER&
+      append_index(size_t i) noexcept {
+        index_.append(i);
+        if (i > max_index_) {
+          max_index_ = i;
+        }
+        return content_;
+      }
```

Comment:
@jpivarski I have implemented tracking max_index_ for Indexed, however I am not checking that the index is >= 0.
The problem is that if I add the check in `append_index` I should then add a `error` argument to store the error message?

Reply:
I think the `>= 0` check could even be ignored. The main reason for the `is_valid` checks is to make sure that non-local constraints are satisfied: it's easy to make one node locally correct and another node locally correct and not notice that they won't work together, so these `is_valid` checks prevent that kind of mistake.

Passing a negative value to an IndexedArray's `index` is a local, more obvious kind of mistake. A non-negative check could be applied immediately when the `index_` is appended to in the first place. If that function is a `noexcept`, then it could set the running `max_index_` to `MAXINT`, so that it would be caught here.

(Thanks for pinging me! Go ahead and do that whenever I've been slow to respond.)

Reply:
ok @jpivarski, implemented that, used `UINTMAX_MAX` for max value of a `size_t`

Reply:
should I check for that value in `is_valid` and explain the reason for the error?

Reply:
Yes, a custom error message for `UINTMAX_MAX`, as opposed to any other `max_index_ >= content_.length()`, would help a user because of its different cause.

(I don't know why I missed this 2 days ago. I'm going to keep this tab open, along with my email, Slacks, and all that, so that I check it in my regular rotation.)

Reply:
@jpivarski ok, implemented this and rebased on main